neo/rawio/spikegadgetsrawio.py

```diff
@@ -103,7 +123,21 @@ def _parse_header(self):
 
         self._sampling_rate = float(hconf.attrib["samplingRate"])
         num_ephy_channels = int(hconf.attrib["numChannels"])
-
+        
+        # check for agreement with number of channels in xml
+        sconf_channels = np.sum([len(x) for x in sconf])
+        if sconf_channels < num_ephy_channels:
+            num_ephy_channels = sconf_channels
+        if sconf_channels > num_ephy_channels:
+            raise ValueError(
+                "SpikeGadgets: the number of channels in the spike configuration is larger than the number of channels in the hardware configuration"
+            )
+
+        try:
+            num_chan_per_chip = int(sconf.attrib["chanPerChip"])
+        except KeyError:
+            num_chan_per_chip = 32  # default value for Intan chips
```

Comment:
Intan also has 64 and 16 channel headstages. Where is this 32 coming from?

Reply:
most spikegadgets products use 64 channel chips, but a 64 channel chip is just two 32 channel chips sharing an output pin so this is a reasonable default for most cases.

Reply:
> most spikegadgets products use 64 channel chips, but a 64 channel chip is just two 32 channel chips sharing an output pin so this is a reasonable default for most cases.
Again, sorry to interrupt... Is this still work in signal channel level? I mean in practice,  does it mean there is no need to change num_chan_per_chip manually into 64 even we know we are using a 64 channel chip, but the header_text xml extracts as `sconf.attrib["chanPerChip"] = 32`?

Reply:
@jnjnnjzch if you are using [this product](https://spikegadgets.com/products/msh-modular-datalogger/) then yes, no need to change `num_chan_per_chip`. If you're using some other product, I'm actually not sure what would happen here...

Reply:
Thanks! we are exactly using this product! And your opinion is validated again by checking the signal synchronization of the extracted trace. Thanks a lot!